kodi_addon_checker/check_addon.py

```diff
@@ -384,3 +389,31 @@ def _check_dependencies(report: Report, addon_path, repo_addons):
             if LooseVersion(available_version) < LooseVersion(required_version) and (required_addon not in ignore):
                 report.add(Record(PROBLEM, "Version mismatch for addon %s. Required: %s, Available: %s "
                                   % (required_addon, required_version, available_version)))
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(report: Report, addon_path, all_repo_addons):
+    """Check if addon submitted already exists or not"""
+    addon_xml = os.path.join(addon_path, "addon.xml")
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    branches = ['gotham', 'helix', 'isengard', 'jarvis', 'krypton', 'leia']
+
+    for branch in branches:
+        repo_addons = all_repo_addons[branch]
+
+        if addon_name in repo_addons:
+            if LooseVersion(addon_version) <= LooseVersion(repo_addons[addon_name]):
+                report.add(Record(PROBLEM, "%s addon already exist with current version in lower branch"
```

Comment:
I would change 'exist with **current version** in **lower branch**' to something mentioning the version and the branch